src/NuGet.Core/NuGet.Protocol/Plugins/PluginCacheEntry.cs

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using NuGet.Common;
+
+namespace NuGet.Protocol.Plugins
+{
+    public sealed class PluginCacheEntry
+    {
+        public PluginCacheEntry(string rootCacheFolder, string pluginFilePath, string requestKey)
+        {
+            RootFolder = Path.Combine(rootCacheFolder, CachingUtility.RemoveInvalidFileNameChars(CachingUtility.ComputeHash(pluginFilePath)));
+            CacheFileName = Path.Combine(RootFolder, CachingUtility.RemoveInvalidFileNameChars(requestKey) + ".dat");
+            NewCacheFileName = CacheFileName + "-new";
+        }
+
+        internal TimeSpan MaxAge { get; set; } = TimeSpan.FromDays(30);
+        internal string CacheFileName { get; }
+        private string RootFolder { get; }
+        private string NewCacheFileName { get; }
+
+        public IReadOnlyList<OperationClaim> OperationClaims { get; private set; }
+
+        public void LoadFromFile()
+        {
+            Stream content = null;
+            try
+            {
+                content = CachingUtility.TryReadCacheFile(MaxAge, CacheFileName);
```

Comment:
APIs that usually start with Try... usually take an out param and return a bool. It's a little weird that CachingUtility.TryReadCacheFile is not doing that

Reply:
All APIs in CachingUtility existed before, I merely extracted them to common plan so I can reuse them. Nice catch :)